src/app/api/user/[userId]/bookAppointment/route.js

```diff
@@ -4,30 +4,39 @@ import { createClient } from "../../../../../../utils/supabase/server";
 export const POST = async(request, { params }) => {
   try {
     const supabase = await createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY)
-
     const body = await request.json();
+    const { userId } = await params;
 
-    console.log(body)
+    const existingUser = await supabase.from("user_details").select("*").eq("id", userId).single();
 
-    const { userId } = await params;
+    if (existingUser.error) return NextResponse.json({error: "A user with this id does not exist "}, {status: 404});
 
-    console.log(userId)
+    const alreadyHasAppointment = await supabase.from("appointments").select("*").eq("resident_id", userId)
 
-    const { data: user, error } = await supabase.from("user_details").select("*").eq("id", userId).single();
+    if (alreadyHasAppointment.data.length > 0) return NextResponse.json({error: `Error: You already have an appointment booked.`}, {status: 403});
 
-    if (error) return NextResponse.json({error: "A user with this id does not exist "}, {status: 404});
+    //Make sure the address is valid
+    const addressInfo = body.address.split(" ");
+    const addressNumber = addressInfo[0]
+    const addressName = addressInfo[1]
 
-    if (user.role === 1) return NextResponse.json({error: "Admins cannot book appointments."}, {status: 403});
+    const checkAddress = await fetch(`https://data.lacity.org/resource/4ca8-mxuh.json?$query=SELECT%20%60hse_nbr%60,%20%60str_nm%60,%20%60str_sfx_cd%60,%20%60zip_cd%60,%20%60lat%60,%20%60lon%60%20WHERE%20%60hse_nbr%60=${addressNumber}%20AND%20LOWER(%60str_nm%60)=LOWER('${addressName}')%20LIMIT%2010`);
+    const validAddress = await checkAddress.json();
 
-    const alreadyHasAppointment = await supabase.from("appointments").select("*").eq("resident_id", userId)
+    if (validAddress.length === 0) {
+      return NextResponse.json({error: "The address submitted is not a valid address in LA. Please try again."}, {status:404})
+    }
 
-    if (alreadyHasAppointment.data.length > 0) return NextResponse.json({error: `Error: You already have an appointment booked.`}, {status: 403});
+    // if (user.role === 1) return NextResponse.json({error: "Admins cannot book appointments."}, {status: 403});

```

Comment:
Nice! I comment out old code too because you never know if you'll need it as a reference for later!

Reply:
I actually need to refactor that lol.